src/java/org/apache/cassandra/security/FileBasedSslContextFactory.java

```diff
@@ -53,6 +54,8 @@ abstract public class FileBasedSslContextFactory extends AbstractSslContextFacto
 
     @VisibleForTesting
     protected volatile boolean checkedExpiry = false;
```

Comment:
At the configuration level, I think it is good to not alter the existing names, since they are user-facing. 

But at the implementation, what do you think about renaming the variables to reflect that they are used for inbound?  e.g. `checkedExpiry` --> `checkedExpiryInboundKeystore`, so the code is more consistent and easier for maintainers to understand. This is suggestion 1.

We can even take a step further to create a data class to avoid field and logic duplications. This is suggestion 2. For example, 
```java
    static class KeyStoreContext
    {
        String file; // keystore
        String password; // keystore_password
        volatile boolean checkExpiry;

        public KeyStoreContext(String file, String password, boolean checkExpiry)
        {
            this.file = file;
            this.password = password;
            this.checkExpiry = checkExpiry;
        }

        // add appropriate method. e.g. a hasKeystore equivelant exist()
    }
```

Whatever suggestion you take, please make sure the other files are also changed.

Reply:
@yifan-c `checkedExpiry` represents not just inboundKeystore expiry in ServerEncryptionOptions, It also represents expiry for the keystore in `EncryptionOptions` which is clientEncryptionOptions. So I think we should not rename it.

Reply:
If `checkedExpiry` is used for both server and client encryption options. Can you help me to understand what is the use of `checkedExpiryOutboundKeystore`? 

The idea of my prior comment was to improve the code clarity as there are a lot of duplications.

Reply:
@yifan-c  following is the class structure of both the classes

EncryptionOptions (Used as client encryption options)
   - keystore
   - password
`checkedExpiry represents`  if keystore got expired for client client connections 

ServerEncryptionOptions extends EncryptionOptions
  - keystore
  - password
  - outboundkeystore
  - outbound password
`checkedExpiry` represents if keystore got expired for internode inbound connections
`checkedExpiryOutboundKeystore` represents if keystore got expired for outbound connections.

So, we should not rename keystore to inbound keystore or checkedExpiry to checkedExpiryInboundKeystore

Reply:
Thanks for implementing the data class to simplify the code a bit.